### PR TITLE
LibGfx: Add scaffolding for curves in lutAToBType and lutBToAType

### PR DESCRIPTION
**Pull request by nico**

### CODE REVIEW

File: Userland/Libraries/LibGfx/ICC/TagTypes.cpp [hunk 1]
```diff
@@ -358,6 +358,17 @@ static ErrorOr<CLUTData> read_clut_data(ReadonlyBytes bytes, AdvancedLUTHeader c
...
 }
 
+static ErrorOr<Vector<LutCurveType>> read_curves(ReadonlyBytes bytes, u32 offset, u32 count)
+{
+    Vector<LutCurveType> curves;
+    // FIXME: Implement.
+    (void)bytes;
+    (void)offset;
+    for (u32 i = 0; i < count; ++i)
+        TRY(curves.try_append(adopt_ref(*new CurveTagData(0, 0, {}))));
```
Comment (awesomekling) [inline Userland/Libraries/LibGfx/ICC/TagTypes.cpp:368]: For extra GBP, you can `adopt_nonnull_own_or_enomem(new (nothrow) CurveTagData...` and also fail with `ENOMEM` here.

Comment (nico) [reply Userland/Libraries/LibGfx/ICC/TagTypes.cpp:368]: I'll make a separate pr to update this whole file to that.

Comment (nico) [reply Userland/Libraries/LibGfx/ICC/TagTypes.cpp:368]: Hmm, looks like there isn't a `adopt_nonnull_ref_or_enomem`, and this uses RefPtrs, not OwnPtrs. Should there be something like that for RefPtrs too?

Comment (nico) [reply Userland/Libraries/LibGfx/ICC/TagTypes.cpp:368]: Ah no, it exists, it's just in RefPtr.h instead of in NonnullRefPtr.h!

Comment (nico) [reply Userland/Libraries/LibGfx/ICC/TagTypes.cpp:368]: -> #17426.

Comment (nico) [reply Userland/Libraries/LibGfx/ICC/TagTypes.cpp:368]: …feel like updating Documentation/SmartPointers.md with when to pick `adopt_nonnull_own_or_enomem` over `adopt_own` (and same for RefPtr)? (I'm guessing it's "always" (?), but the doc currently doesn't mention these at all and recommend the old tech.)

Comment (nico) [reply Userland/Libraries/LibGfx/ICC/TagTypes.cpp:368]: II opened #17432 for the documentation needing an update.

**Pull request merged**